src/controllers/todo-item.controller.ts

```diff
@@ -36,39 +36,39 @@ export class TodoItemController {
   @param.body('todoItem', TodoItemSchema)
   async createTodoItem(id: number, todoItem: TodoItem) {
     todoItem.todoId = id;
-    let todo;
-    try {
-      todo = await this.todoRepo.findById(id);
-    } catch (err) {
+    const todo = await this.todoRepo.findById(id);
+    if (!todo) {
       return Promise.reject(
         new HttpErrors.NotFound(`No todo was found with id ${id}`)
       );
     }
-    if (todo) {
-      return await this.todoItemRepo.create(todoItem);
-    }
+    return await this.todoItemRepo.create(todoItem);
   }
 
   @put('/todo/{x}/items/{id}')
   @put('/todoitem/{id}')
   @param.path.number('id')
   @param.body('todoItem', TodoItemSchema)
   @param.path.number('x')
-  async replaceTodoItem(id: number, item: TodoItem, x?: number) {
-    return await this.todoItemRepo.replaceById(id, item);
+  async replaceTodoItem(id: number, todoItem: TodoItem, x?: number) {
+    const item = await this.todoItemRepo.findById(id);
+    if (!item) {
+      return Promise.reject(
+        new HttpErrors.NotFound(`No todo item was found with id ${id}`)
+      );
+    }
+    return await this.todoItemRepo.deleteById(id, todoItem);
   }
 
   @patch('/todo/{x}/items/{id}')
   @patch('/todoitem/{id}')
   @param.path.number('x')
   @param.body('todoItem', TodoItemSchema)
   @param.path.number('id')
-  async updateTodoItem(x: number, todoItem: TodoItem, id: number) {
+  async updateTodoItem(id: number, todoItem: TodoItem, x?: number) {
     // We don't need the todoId, but it's for consistency.
-    let item;
-    try {
-      item = await this.todoItemRepo.findById(id);
-    } catch (err) {
+    const item = await this.todoItemRepo.findById(id);
```

Comment:
This pattern where each controller method calls `findById` first and then performs the operation, is creating a race condition. For example, what if another clients deletes the todoItem instance after `findById` was finished but before `updateById` below was called?

IMO, the controller methods should not be calling `findById` at all. It's the job of the database connector (or the juggler DAO layer) to return a meaningful error when we are attempting to edit or delete an object that does not exist. Only the connector is able to check for existence and modify/delete the model instance in an atomic way.